pkg/util/errors.go

```diff
@@ -36,12 +35,13 @@ func NewStillExists(qualifiedResource schema.GroupResource, name string) *apierr
 	}
 }
 
-func IsConnectionRefusedOrTimeout(err error) bool {
-	urlErr := err.(*url.Error)
-	if urlErr.Timeout() {
+func IsRetriableNetworkError(err error) bool {
+	if errors.Is(err, syscall.ETIMEDOUT) {
 		return true
 	} else if errors.Is(err, syscall.ECONNREFUSED) {
 		return true
+	} else if errors.Is(err, syscall.EHOSTUNREACH) {
+		return true
```

Comment:
we may consider add those erorrs also into retryiable network error
```
ENETUNREACH
ENETDOWN
ECONNRESET
```

Reply:
Updated, thanks.